src/components/OverviewMenu/OverviewMenu.tsx

```diff
@@ -0,0 +1,217 @@
+        className={
+          paths.circles?.includes(location.pathname) ||
+          location.pathname.includes(paths.history)
+            ? 'active'
+            : ''
+        }
+        href="#"
+      >
+        {overviewMenuTriggerText}
+        <Box css={{ marginLeft: '$xs', display: 'flex' }}>
+          <ChevronDown size="md" />
+        </Box>
+      </Link>
+      <Hidden smDown>
+        <Popover
+          anchorEl={anchorEl}
+          classes={{ paper: classes.popover }}
+          id="overview-popover"
+          onClick={() => setTimeout(() => setAnchorEl(null))}
+          onClose={() => setAnchorEl(null)}
+          open={!!anchorEl}
+        >
+          <Box
+            css={{
+              display: 'flex',
+              flexDirection: 'column',
+              p: '$md',
+            }}
+          >
+            <Link
+              type="menu"
+              css={{
+                py: '$sm',
+                fontWeight: '$bold',
+                display: 'flex',
+                flexDirection: 'row',
+                alignItems: 'center',
+              }}
+            >
+              {overviewMenuTriggerText}
+              <Box css={{ marginLeft: '$xs', display: 'flex' }}>
+                <ChevronUp size="md" />
+              </Box>
+            </Link>
+            <Box
+              css={{
+                display: 'flex',
+                flexDirection: 'column',
+                marginTop: '$sm',
+              }}
+            >
+              {hasCircles && <TopLevelLinks links={mainLinks} />}
+            </Box>
+            {orgs?.map(org => (
+              <Box key={org.id} css={menuGroup}>
+                <Text variant="label" as="label">
+                  {org.name}
+                </Text>
+                <Box css={{ display: 'flex', flexDirection: 'column' }}>
+                  {sortBy(org.circles, c => -c.users.length).map(circle => (
+                    <CircleItem
+                      circle={circle}
+                      key={circle.id}
+                      onButtonClick={goToCircle}
+                    />
+                  ))}
+                </Box>
+              </Box>
+            ))}
+          </Box>
+        </Popover>
+      </Hidden>
+    </>
+  );
+};
+
+type QueryCircle = QueryResult['organizations'][0]['circles'][0];
+
+type CircleItemProps = {
+  circle: QueryCircle;
+  onButtonClick: (id: number, path: string) => void;
+};
+
+export const TopLevelLinks = ({
+  links,
+}: {
+  links: [string, string, string[]?][];
+}) => {
+  const location = useLocation();
+
+  return (
+    <>
+      {links.map(([path, label, matchPaths]) => (
+        <Link
+          type="menu"
+          css={{ pt: '$sm' }}
+          as={NavLink}
+          key={path}
+          to={path}
+          className={matchPaths?.includes(location.pathname) ? 'active' : ''}
+        >
+          {label}
+        </Link>
+      ))}
+    </>
+  );
+};
```

Comment:
Would be good to pull these out into separate component files to keep things more modular

Reply:
I'd like to punt on moving this out to another component for the time being.  The MainHeader component has something similar, but it seems like a good deal of complexity to refactor it neatly.